packages/shared/src/components/cards/common/CardCover.tsx

```diff
@@ -26,15 +25,18 @@ export function CardCover({
   const { shouldUseMobileFeedLayout } = useFeedLayout();
   const ImageComponent = shouldUseMobileFeedLayout ? CardImageV1 : CardImage;
   const { shouldShowOverlay, onInteract } = usePostShareLoop(post);
-  const coverShare = (
-    <CardCoverShare
-      post={post}
-      onShare={() => {
-        onInteract();
-        onShare(post);
-      }}
-      onCopy={onInteract}
-    />
+  const coverShare = useMemo(
+    () => (
+      <CardCoverShare
+        post={post}
+        onShare={() => {
+          onInteract();
+          onShare(post);
+        }}
+        onCopy={onInteract}
+      />
+    ),
+    [onInteract, onShare, post],
```

Comment:
No point in memoizing this. React has no use of this memozation and it is super fast in parsing virtual dom. You can move it below inline. As you seen in this issue, actually writting to DOM is the heaviest operation.

Reply:
Yes, this was an oversight of mine, I tried a different approach before and forgot to revert this part.